src/SimpleEventStore/SimpleEventStore/InMemory/InMemoryStorageEngine.cs

```diff
@@ -26,7 +26,7 @@ public Task AppendToStream(string streamId, IEnumerable<StorageEvent> events, Ca
 
                 if (firstEvent.EventNumber - 1 != streams[streamId].Count)
                 {
-                    throw new ConcurrencyException($"Concurrency conflict when appending to stream {streamId}. Expected revision {firstEvent.EventNumber} : Actual revision {streams[streamId].Count}");
+                    throw new ConcurrencyException($"Concurrency conflict when appending to stream {streamId}. Expected revision {firstEvent.EventNumber - 1}");
```

Comment:
Ideally the expected version should be passed into the method so we aren't expecting each storage engine to query the first event number - 1.  However, this would be a breaking change to the API, so this will be done as part of a subsequent PR.